sys/include/xtimer/tick_conversion.h

```diff
@@ -29,16 +29,135 @@ extern "C" {
 #endif
 
 /* Some optimizations for common timer frequencies */
+
+/*
+ * First, check if XTIMER_HZ is multiple of 921600Hz
+ * This is a common frequency for baud rate generation. We use the fact that
+ * the greatest common divisor between 921600 and 1000000 is 1600, so instead of
+ * multiplying by the fraction (921600 / 1000000), we will instead use
+ * (576 / 625), which reduces the truncation caused by the integer widths
+ */
+#if (XTIMER_HZ % 576 == 0)
+/*
+ * Second, check if it's using shifting
+ */
 #if (XTIMER_SHIFT != 0)
+/*
+ * Since we are using 921600 as a base, first check if the given frequency is
+ * greater than it.
+ */
+#if (XTIMER_HZ > 921600ul)
+/*
+ * If the shift on XTIMER_HZ doesn't match with the current speed, throw an
+ * error
+ */
+#if (XTIMER_HZ != (921600ul << XTIMER_SHIFT))
+#error XTIMER_HZ != (921600ul << XTIMER_SHIFT)
+#endif
+/*
+ * Use functions to return an equivalent conversion to XTIMER_SHIFT shifts for
+ * high frequencies
+ */
+inline static uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
+    return div_u32_by_625div576(usec) << XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
+    return div_u64_by_625div576(usec) << XTIMER_SHIFT;
+}
+
+inline static uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
+    return div_u32_by_576div625(ticks) >> XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
+    return div_u64_by_576div625(ticks) >> XTIMER_SHIFT;
+}
+/*
+ * If it's not greater, check if it's lower (XTIMER_HZ < 921600ul).
+ */
+#else
+/*
+ * Again, if the shifts doesn't match with 921600, throw an error.
+ */
+#if ((XTIMER_HZ << XTIMER_SHIFT) != 921600ul)
+#error (XTIMER_HZ << XTIMER_SHIFT) != 921600ul
+#endif
+/*
+ * Use functions to return an equivalent conversion to XTIMER_SHIFT shifts for
+ * low frequencies.
+ */
+inline static uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
+    return div_u32_by_625div576(usec) >> XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
+    return div_u64_by_625div576(usec) >> XTIMER_SHIFT;
+}
+
+inline static uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
+    return div_u32_by_576div625(ticks) << XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
+    return div_u64_by_576div625(ticks) << XTIMER_SHIFT;
+}
+/*
+ * No more shifted multiple of 921600Hz frequencies to check
+ * (XTIMER_SHIFT != 0)
+ */
+#endif
+/*
+ * Check for the actual frequency without any shifting (921600Hz)
+ * (XTIMER_SHIFT == 0)
+ */
+#elif (XTIMER_HZ == 921600ul)
```

Comment:
I don't think you need a special case for that, shifting to any side by 0 (`XTIMER_SHIFT` should be 0 here). Would give the same result.